## Final Project Submission

Please fill out:
* Student name: 
* Student pace: self paced / part time / full time
* Scheduled project review date/time: 
* Instructor name: 
* Blog post URL:


In [21]:
!ls

CONTRIBUTING.md
LICENSE.md
README.md
awesome.gif
data.sqlite
im.db
movie_data_erd.jpeg
student.ipynb
zippedData


# Importing the relevant libraries
Begin by importing pandas, numpy, csv with their aliases

In [22]:
# Your code here - remember to use markdown cells for comments as well!
#importing libraries
import pandas as pd
import sqlite3
import csv
import numpy as np
from datetime import datetime


# Exploring the datasets
Preview the first dataset, which is from "tn.movie_budgets.csv", which is inside the zipped data

In [48]:



#Load the data from tn.movie_budgets.csv into a dataframe
movies_budget = pd.read_csv('zippedData/tn.movie_budgets.csv')

#Cast to numeric data type
movies_budget['production_budget'] = movies_budget['production_budget'].replace('[^.0-9]', '', regex=True).astype('float32')
movies_budget['domestic_gross'] = movies_budget['domestic_gross'].replace('[^.0-9]', '', regex=True).astype('float32')
movies_budget['worldwide_gross'] = movies_budget['worldwide_gross'].replace('[^.0-9]', '',regex = True).astype('float64')
movies_budget.head(5)

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross  
0        425000000.0     760507648.0     2.776345e+09  
1        410600000.0     241063872.0     1.045664e+09  
2        350000000.0      42762352.0     1.497624e+08  
3        330600000.0     459005856.0     1.403014e+09  
4        317000000.0     620181376.0     1.316722e+09

The production budget, domestic gross, and worldwide gross are then cast to float type for mathematical manipulation.

In [49]:
movies_budget.shape

(5782, 6)

In [50]:
movies_budget.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5782 non-null   int64  
 1   release_date       5782 non-null   object 
 2   movie              5782 non-null   object 
 3   production_budget  5782 non-null   float32
 4   domestic_gross     5782 non-null   float32
 5   worldwide_gross    5782 non-null   float64
dtypes: float32(2), float64(1), int64(1), object(2)
memory usage: 226.0+ KB


In [51]:
movies_budget.isna().sum()

id                   0
release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

The movie budget Data frame has 5,782 rows and 6 columns namely: id,release_date, movie, production_budget, domestic gross and worldwide_gross. The data types are as float,object and int. The are no null values.

Finding out the movies budget, first we get the international gross by subtracting the domestic gross from the woorldwide gross

In [52]:
#getting the international gross
movies_budget['international_gross'] = movies_budget['worldwide_gross'] - movies_budget['domestic_gross']
cols = list(movies_budget.columns)

movies_budget.head(10)

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
5   6  Dec 18, 2015         Star Wars Ep. VII: The Force Awakens   
6   7  Apr 27, 2018                       Avengers: Infinity War   
7   8  May 24, 2007   Pirates of the Caribbean: At World’s End   
8   9  Nov 17, 2017                               Justice League   
9  10   Nov 6, 2015                                      Spectre   

   production_budget  domestic_gross  worldwide_gross  international_gross  
0        425000000.0     760507648.0     2.776345e+09         2.015838e+09  
1        410600000.0     241063872.0     1.045664e+09         8.046000e+08  
2        350000000.0      42762352.0     1.497624e+08         1.070000e+08  
3        330600000.0     459005856.0     1.403014e+09         9.440081e+08  
4        317000000.0     620181376.0     1.316722e+09         6.965404e+08  
5        306000000.0     936662208.0     2.053311e+09         1.116649e+09  
6        300000000.0     678815488.0     2.048134e+09         1.369319e+09  
7        300000000.0     309420416.0     9.634204e+08         6.540000e+08  
8        300000000.0     229024288.0     6.559452e+08         4.269209e+08  
9        300000000.0     200074176.0     8.796209e+08         6.795467e+08

The release date is an object, in order to filter the years convert to date time type.

In [53]:
#converting object to dateTime
movies_budget['release_date'] = pd.to_datetime(movies_budget['release_date'], errors='coerce')
movies_budget['year'] = movies_budget['release_date'].dt.strftime('%Y')
movies_budget.head(10)

id release_date                                        movie  \
0   1   2009-12-18                                       Avatar   
1   2   2011-05-20  Pirates of the Caribbean: On Stranger Tides   
2   3   2019-06-07                                 Dark Phoenix   
3   4   2015-05-01                      Avengers: Age of Ultron   
4   5   2017-12-15            Star Wars Ep. VIII: The Last Jedi   
5   6   2015-12-18         Star Wars Ep. VII: The Force Awakens   
6   7   2018-04-27                       Avengers: Infinity War   
7   8   2007-05-24   Pirates of the Caribbean: At World’s End   
8   9   2017-11-17                               Justice League   
9  10   2015-11-06                                      Spectre   

   production_budget  domestic_gross  worldwide_gross  international_gross  \
0        425000000.0     760507648.0     2.776345e+09         2.015838e+09   
1        410600000.0     241063872.0     1.045664e+09         8.046000e+08   
2        350000000.0      42762352.0     1.497624e+08         1.070000e+08   
3        330600000.0     459005856.0     1.403014e+09         9.440081e+08   
4        317000000.0     620181376.0     1.316722e+09         6.965404e+08   
5        306000000.0     936662208.0     2.053311e+09         1.116649e+09   
6        300000000.0     678815488.0     2.048134e+09         1.369319e+09   
7        300000000.0     309420416.0     9.634204e+08         6.540000e+08   
8        300000000.0     229024288.0     6.559452e+08         4.269209e+08   
9        300000000.0     200074176.0     8.796209e+08         6.795467e+08   

   year  
0  2009  
1  2011  
2  2019  
3  2015  
4  2017  
5  2015  
6  2018  
7  2007  
8  2017  
9  2015

In [54]:
movies_budget.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 8 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   id                   5782 non-null   int64         
 1   release_date         5782 non-null   datetime64[ns]
 2   movie                5782 non-null   object        
 3   production_budget    5782 non-null   float32       
 4   domestic_gross       5782 non-null   float32       
 5   worldwide_gross      5782 non-null   float64       
 6   international_gross  5782 non-null   float64       
 7   year                 5782 non-null   object        
dtypes: datetime64[ns](1), float32(2), float64(2), int64(1), object(2)
memory usage: 316.3+ KB


# Filtering the Data
Here we will drop the rows without the values for the domestic, international and worldwide gross.

In [55]:
movies_budget.sort_values('worldwide_gross')

id release_date                                    movie  \
5037  38   2019-04-23  Living Dark: The Story of Ted the Caver   
3975  76   2015-05-15                           Pound of Flesh   
4627  28   2011-06-28                                     2:13   
4628  29   2013-01-29  Batman: The Dark Knight Returns, Part 2   
3947  48   2019-06-21                           Burn Your Maps   
...   ..          ...                                      ...   
33    34   2015-06-12                           Jurassic World   
6      7   2018-04-27                   Avengers: Infinity War   
5      6   2015-12-18     Star Wars Ep. VII: The Force Awakens   
42    43   1997-12-19                                  Titanic   
0      1   2009-12-18                                   Avatar   

      production_budget  domestic_gross  worldwide_gross  international_gross  \
5037          1750000.0             0.0     0.000000e+00         0.000000e+00   
3975          7500000.0             0.0     0.000000e+00         0.000000e+00   
4627          3500000.0             0.0     0.000000e+00         0.000000e+00   
4628          3500000.0             0.0     0.000000e+00         0.000000e+00   
3947          8000000.0             0.0     0.000000e+00         0.000000e+00   
...                 ...             ...              ...                  ...   
33          215000000.0     652270656.0     1.648855e+09         9.965842e+08   
6           300000000.0     678815488.0     2.048134e+09         1.369319e+09   
5           306000000.0     936662208.0     2.053311e+09         1.116649e+09   
42          200000000.0     659363968.0     2.208208e+09         1.548844e+09   
0           425000000.0     760507648.0     2.776345e+09         2.015838e+09   

      year  
5037  2019  
3975  2015  
4627  2011  
4628  2013  
3947  2019  
...    ...  
33    2015  
6     2018  
5     2015  
42    1997  
0     2009  

[5782 rows x 8 columns]

In [56]:
#drop all all rows without the gross values
movies_budget_drop = movies_budget[(movies_budget['domestic_gross'] == 0) & (movies_budget['worldwide_gross'] == 0)]

In [57]:
movies_budget.drop(movies_budget_drop.index, axis=0, inplace=True)
movies_budget.head(5)

id release_date                                        movie  \
0   1   2009-12-18                                       Avatar   
1   2   2011-05-20  Pirates of the Caribbean: On Stranger Tides   
2   3   2019-06-07                                 Dark Phoenix   
3   4   2015-05-01                      Avengers: Age of Ultron   
4   5   2017-12-15            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross  international_gross  \
0        425000000.0     760507648.0     2.776345e+09         2.015838e+09   
1        410600000.0     241063872.0     1.045664e+09         8.046000e+08   
2        350000000.0      42762352.0     1.497624e+08         1.070000e+08   
3        330600000.0     459005856.0     1.403014e+09         9.440081e+08   
4        317000000.0     620181376.0     1.316722e+09         6.965404e+08   

   year  
0  2009  
1  2011  
2  2019  
3  2015  
4  2017

In order to have relevant budgeting information, filter the years to get only the data from 2011

In [58]:
#convert from object to string
movies_budget['year'] = movies_budget['year'].astype(int)


In [59]:
#drop rows which are not between the years 2015 and 2022
drop_=  movies_budget[(movies_budget['year'] <= 2014)]
movies_budget.drop(drop_.index, axis=0, inplace=True)
movies_budget.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 797 entries, 2 to 5748
Data columns (total 8 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   id                   797 non-null    int64         
 1   release_date         797 non-null    datetime64[ns]
 2   movie                797 non-null    object        
 3   production_budget    797 non-null    float32       
 4   domestic_gross       797 non-null    float32       
 5   worldwide_gross      797 non-null    float64       
 6   international_gross  797 non-null    float64       
 7   year                 797 non-null    int32         
dtypes: datetime64[ns](1), float32(2), float64(2), int32(1), int64(1), object(1)
memory usage: 46.7+ KB


In [60]:
movies_budget['year'].value_counts()

2015    242
2016    204
2017    162
2018    140
2019     49
Name: year, dtype: int64

In [61]:
movies_budget.describe()

id  production_budget  domestic_gross  worldwide_gross  \
count  797.000000              797.0           797.0     7.970000e+02   
mean    50.634881         44454288.0      57354036.0     1.477667e+08   
std     29.339221         57002492.0      98196584.0     2.607258e+08   
min      1.000000            25000.0             0.0     2.600000e+01   
25%     25.000000          8000000.0       3000342.0     8.200000e+06   
50%     50.000000         22000000.0      24431472.0     4.517374e+07   
75%     78.000000         55000000.0      63859436.0     1.594876e+08   
max    100.000000        350000000.0     936662208.0     2.053311e+09   

       international_gross         year  
count         7.970000e+02   797.000000  
mean          9.041263e+07  2016.435383  
std           1.710463e+08     1.255101  
min          -1.000000e+00  2015.000000  
25%           2.053330e+06  2015.000000  
50%           1.928622e+07  2016.000000  
75%           9.237100e+07  2017.000000  
max           1.369319e+09  2019.000000

Finding the percentage profits of the movies based on the production budget and the world wide gross.

In [62]:
movies_budget['profit/loss'] = (movies_budget['worldwide_gross'] - movies_budget['production_budget'])
movies_budget['profit/loss'] = movies_budget['profit/loss'].astype(int)      
movies_budget['profit/loss'].tail()

5701    -66201
5719    -41626
5737    114822
5744    -23453
5748    464792
Name: profit/loss, dtype: int32

In [63]:
movies_budget.shape

(797, 9)

The bom dataset will gives information on the domestic gross values and the foreign gross.

In [69]:
#previewing the second dataset
movies_gross = pd.read_csv('zippedData/bom.movie_gross.csv')
movies_gross.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [70]:
#find out the shape of the dataset
movies_gross.shape

(3387, 5)

In [71]:
movies_gross['year'].unique()

array([2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018], dtype=int64)

In [72]:
movies_gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [73]:
movies_gross.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [78]:
#drop the rows without gross values
movies_gross.dropna(subset=['foreign_gross','domestic_gross'], inplace=True)
movies_gross.isna().sum()

title             0
studio            2
domestic_gross    0
foreign_gross     0
year              0
dtype: int64

The tmdb.movies dataset gives us information on the popularity of a movie with respect to the language.

In [79]:
tmd = pd.read_csv('zippedData/tmdb.movies.csv',index_col=0)
tmd.head(5)

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   
3      [16, 35, 10751]    862                en   
4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [83]:
tmd['original_language'].value_counts().head(5)

en    23291
fr      507
es      455
ru      298
ja      265
Name: original_language, dtype: int64

In [84]:
tmd.shape

(26517, 9)

The dataset has 26,517 rows and 9 columns

In [86]:
tmd.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26517 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          26517 non-null  object 
 1   id                 26517 non-null  int64  
 2   original_language  26517 non-null  object 
 3   original_title     26517 non-null  object 
 4   popularity         26517 non-null  float64
 5   release_date       26517 non-null  object 
 6   title              26517 non-null  object 
 7   vote_average       26517 non-null  float64
 8   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 2.0+ MB


In [87]:
tmd.duplicated().sum()

1020

In [99]:
#drop all duplicate rows based on the titles columns
tmd = tmd.drop_duplicates(subset = ['title','original_title'], keep = 'first')
tmd.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 24688 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          24688 non-null  object 
 1   id                 24688 non-null  int64  
 2   original_language  24688 non-null  object 
 3   original_title     24688 non-null  object 
 4   popularity         24688 non-null  float64
 5   release_date       24688 non-null  object 
 6   title              24688 non-null  object 
 7   vote_average       24688 non-null  float64
 8   vote_count         24688 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 1.9+ MB


In [100]:
tmd.duplicated().sum()

0

The data has no duplicates 

In [101]:
tmd['popularity'].max()


80.773

In [102]:
reviews = pd.read_csv('zippedData/rt.reviews.tsv', delimiter='\t',encoding='unicode_escape')
reviews.head(10)

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   
5   3  ... Cronenberg's Cosmopolis expresses somethin...    NaN   fresh   
6   3  Quickly grows repetitive and tiresome, meander...      C  rotten   
7   3  Cronenberg is not a director to be daunted by ...    2/5  rotten   
8   3  Cronenberg's cold, exacting precision and emot...    NaN   fresh   
9   3  Over and above its topical urgency or the bit ...    NaN   fresh   

            critic  top_critic           publisher                date  
0       PJ Nabarro           0     Patrick Nabarro   November 10, 2018  
1   Annalee Newitz           0             io9.com        May 23, 2018  
2     Sean Axmaker           0    Stream on Demand     January 4, 2018  
3    Daniel Kasman           0                MUBI   November 16, 2017  
4              NaN           0        Cinema Scope    October 12, 2017  
5  Michelle Orange           0    Capital New York  September 11, 2017  
6   Eric D. Snider           0     EricDSnider.com       July 17, 2013  
7     Matt Kelemen           0  Las Vegas CityLife      April 21, 2013  
8     Sean Axmaker           0       Parallax View      March 24, 2013  
9     Kong Rithdee           0        Bangkok Post       March 4, 2013

In [103]:
reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


In [104]:
reviews.isna().sum()

id                0
review         5563
rating        13517
fresh             0
critic         2722
top_critic        0
publisher       309
date              0
dtype: int64

In [105]:
reviews.duplicated().sum()

9

In [106]:
reviews.dropna(axis=0, inplace = True)
reviews.head(10)

id                                             review rating   fresh  \
0    3  A distinctly gallows take on contemporary fina...    3/5   fresh   
6    3  Quickly grows repetitive and tiresome, meander...      C  rotten   
7    3  Cronenberg is not a director to be daunted by ...    2/5  rotten   
11   3  While not one of Cronenberg's stronger films, ...     B-   fresh   
12   3  Robert Pattinson works mighty hard to make Cos...    2/4  rotten   
13   3  The anger over the injustice of the financial ...      B   fresh   
14   3  For those who like their Cronenberg thick and ...    3/4   fresh   
15   3  For better or worse - often both - Cosmopolis ...    3/5   fresh   
16   3  For one of the smartest films I've seen in a w...    4/5   fresh   
17   3      Its major problem is that it's not cinematic.      B   fresh   

             critic  top_critic                  publisher                date  
0        PJ Nabarro           0            Patrick Nabarro   November 10, 2018  
6    Eric D. Snider           0            EricDSnider.com       July 17, 2013  
7      Matt Kelemen           0         Las Vegas CityLife      April 21, 2013  
11     Emanuel Levy           0            EmanuelLevy.Com    February 3, 2013  
12   Christian Toto           0              Big Hollywood    January 15, 2013  
13     Robert Roten           0        Laramie Movie Scope     January 7, 2013  
14      Marty Mapes           0                Movie Habit    October 20, 2012  
15        Adam Ross           0             The Aristocrat  September 27, 2012  
16    Patrick Kolan           0             Shotgun Cinema  September 26, 2012  
17  Dennis Schwartz           0  Ozus' World Movie Reviews  September 25, 2012

From the reviews dataframe it is impossible to get valid data that can be used to draw conclusion on the ratings and the reviews

# checking data from im.db

In [107]:
conn = sqlite3.connect('zippedData/im.db')
cur = conn.cursor()

# Find the table names in im.db

In [108]:
cur.execute("""SELECT name FROM sqlite_master WHERE type = 'table';""")
# Fetch the result and store it in table_names
table_names = cur.fetchall()
table_names

[('movie_basics',),
 ('directors',),
 ('known_for',),
 ('movie_akas',),
 ('movie_ratings',),
 ('persons',),
 ('principals',),
 ('writers',)]

Im.db has 8 tables. The movie_ratings and movie_basics are more relevant to our study as they will help gain information on the highly rated genres.

In [109]:
ratings = pd.read_sql_query("""
SELECT movie_id, start_year, runtime_minutes, genres, averagerating, numvotes
FROM movie_basics
JOIN movie_ratings
    USING(movie_id)
WHERE start_year >= 201
LIMIT 10
;""", conn)

ratings

movie_id  start_year  runtime_minutes                      genres  \
0  tt0063540        2013            175.0          Action,Crime,Drama   
1  tt0066787        2019            114.0             Biography,Drama   
2  tt0069049        2018            122.0                       Drama   
3  tt0069204        2018              NaN                Comedy,Drama   
4  tt0100275        2017             80.0        Comedy,Drama,Fantasy   
5  tt0112502        2017              NaN             Horror,Thriller   
6  tt0137204        2017             83.0  Adventure,Animation,Comedy   
7  tt0146592        2010            136.0                       Drama   
8  tt0154039        2010            100.0                     History   
9  tt0159369        2013            180.0                 Documentary   

   averagerating  numvotes  
0            7.0        77  
1            7.2        43  
2            6.9      4517  
3            6.1        13  
4            6.5       119  
5            4.1        32  
6            8.1       263  
7            6.8       451  
8            4.6        64  
9            7.6        53

In [110]:
movie_akas = pd.read_sql_query("""SELECT * 
FROM movie_akas
;""",conn)

movie_akas.tail(10)

movie_id  ordering                          title region language  \
331693  tt9726638         3  Qi Tian Da Sheng Huo Yan Shan     CN      yue   
331694  tt9755806         1                      Big Shark     US     None   
331695  tt9755806         2                  Большая Акула     RU     None   
331696  tt9755806         3                      Big Shark   None     None   
331697  tt9827784         1             Sayonara kuchibiru     JP     None   
331698  tt9827784         2             Sayonara kuchibiru   None     None   
331699  tt9827784         3                  Farewell Song    XWW       en   
331700  tt9880178         1                    La atención   None     None   
331701  tt9880178         2                    La atención     ES     None   
331702  tt9880178         3                  The Attention    XWW       en   

              types attributes  is_original_title  
331693  imdbDisplay       None                0.0  
331694         None       None                0.0  
331695         None       None                0.0  
331696     original       None                1.0  
331697         None       None                0.0  
331698     original       None                1.0  
331699  imdbDisplay       None                0.0  
331700     original       None                1.0  
331701         None       None                0.0  
331702  imdbDisplay       None                0.0

In [111]:
#find the languages 
movie_akas['language'].unique()
movie_akas['language'].value_counts().head(10)

en     22895
tr      3847
bg      3609
fr      3576
he      2680
sv       965
cmn      727
fa       482
hi       307
yue      286
Name: language, dtype: int64

In [112]:
movie_info = pd.read_csv('zippedData/rt.movie_info.tsv', delimiter ='\t', encoding='unicode_escape' )
movie_info

id                                           synopsis rating  \
0        1  This gritty, fast-paced, and innovative police...      R   
1        3  New York City, not-too-distant-future: Eric Pa...      R   
2        5  Illeana Douglas delivers a superb performance ...      R   
3        6  Michael Douglas runs afoul of a treacherous su...      R   
4        7                                                NaN     NR   
...    ...                                                ...    ...   
1555  1996  Forget terrorists or hijackers -- there's a ha...      R   
1556  1997  The popular Saturday Night Live sketch was exp...     PG   
1557  1998  Based on a novel by Richard Powell, when the l...      G   
1558  1999  The Sandlot is a coming-of-age story about a g...     PG   
1559  2000  Suspended from the force, Paris cop Hubert is ...      R   

                                                  genre            director  \
0                   Action and Adventure|Classics|Drama    William Friedkin   
1                     Drama|Science Fiction and Fantasy    David Cronenberg   
2                     Drama|Musical and Performing Arts      Allison Anders   
3                            Drama|Mystery and Suspense      Barry Levinson   
4                                         Drama|Romance      Rodney Bennett   
...                                                 ...                 ...   
1555   Action and Adventure|Horror|Mystery and Suspense                 NaN   
1556                 Comedy|Science Fiction and Fantasy        Steve Barron   
1557  Classics|Comedy|Drama|Musical and Performing Arts      Gordon Douglas   
1558    Comedy|Drama|Kids and Family|Sports and Fitness  David Mickey Evans   
1559  Action and Adventure|Art House and Internation...                 NaN   

                                                writer  theater_date  \
0                                       Ernest Tidyman   Oct 9, 1971   
1                         David Cronenberg|Don DeLillo  Aug 17, 2012   
2                                       Allison Anders  Sep 13, 1996   
3                      Paul Attanasio|Michael Crichton   Dec 9, 1994   
4                                         Giles Cooper           NaN   
...                                                ...           ...   
1555                                               NaN  Aug 18, 2006   
1556  Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner  Jul 23, 1993   
1557                                               NaN   Jan 1, 1962   
1558                  David Mickey Evans|Robert Gunter   Apr 1, 1993   
1559                                        Luc Besson  Sep 27, 2001   

          dvd_date currency  box_office      runtime             studio  
0     Sep 25, 2001      NaN         NaN  104 minutes                NaN  
1      Jan 1, 2013        $     600,000  108 minutes  Entertainment One  
2     Apr 18, 2000      NaN         NaN  116 minutes                NaN  
3     Aug 27, 1997      NaN         NaN  128 minutes                NaN  
4              NaN      NaN         NaN  200 minutes                NaN  
...            ...      ...         ...          ...                ...  
1555   Jan 2, 2007        $  33,886,034  106 minutes    New Line Cinema  
1556  Apr 17, 2001      NaN         NaN   88 minutes  Paramount Vantage  
1557  May 11, 2004      NaN         NaN  111 minutes                NaN  
1558  Jan 29, 2002      NaN         NaN  101 minutes                NaN  
1559  Feb 11, 2003      NaN         NaN   94 minutes  Columbia Pictures  

[1560 rows x 12 columns]

In [ ]:
movie_info.info()